### PR TITLE
Build + improvements

### PR DESCRIPTION
**Pull request by icecrime**
Excluding `extensions/simplebridge/`, the branch now builds.

Changes:
- Cleanup in `daemon.Create`
- Fist draft of an `extensions.Controller`
- **Interface change** `network.Driver.Unlink()` now takes a `sandbox.Sandbox` parameter: like we expect `Link` to introduce new interfaces in the sandbox, it is likely that `Unlink` will want to _remove_ things from the sandbox.
- **Interface change** Introduce `network.Endpoint.Expose()`: it takes a port specification as a `string` (like `nat.Port`, i.e.: `"80/tcp"`)

### CODE REVIEW

File: daemon/create.go [hunk 1]
```diff
@@ -134,44 +167,16 @@ func (daemon *Daemon) Create(config *runconfig.Config, hostConfig *runconfig.Hos
...
-		if err := daemon.execdriver.AddIface(iface, c.ID); err != nil {
-			return nil, nil, err
-		}
-	}
-
 	// Expose ports on the new endpoint
 	if c.Config.ExposedPorts != nil {
-		for _, port := range config.ExposedPorts {
-			if err := ep.Expose(port, false); err != nil {
+		for port, _ := range config.ExposedPorts {
```
Comment (erikh) [inline daemon/create.go:178]: this can be simplified as: 

```
for port := range config.ExposedPorts
```

File: daemon/create.go [hunk 2]
```diff
@@ -180,11 +185,15 @@ func (daemon *Daemon) Create(config *runconfig.Config, hostConfig *runconfig.Hos
...
+			}
+
+			// FIXME:networking Port is published at the network driver level,
+			// but we are missing the host behaviour: should we rely on
+			// daemon/networkdriver/portmapper/ in a first version?
 			for _, bb := range b {
-				if err := ep.Expose(p, true); err != nil {
-					return nil, nil, err
-				}
+				_ = bb
```
Comment (erikh) [inline daemon/create.go:196]: what is this for?

Comment (icecrime) [inline daemon/create.go:196]: Just a placeholder so it could build :p

File: daemon/create.go (hunk 1 above)
Comment (icecrime) [inline daemon/create.go:178]: Oh right, thanks!

**Pull request merged**